### PR TITLE
Egamma and Muon pre-recommendations AND Muon Trigger Matching

### LINKED ISSUE #110: Perform Trigger Matching for Muons
See [TWiki](https://twiki.cern.ch/twiki/bin/view/AtlasProtected/MCPAnalysisGuidelinesMC15#Muon_trigger_matching).

### PR DESCRIPTION
**Pull request by mmilesi**
This is an important update for muons and electrons.
All the changes were developed and tested in 2.3.14. For what I can see, it works fine.
however, I encourage people to have a careful look at the changes first.

This would close #110 and #102

### CODE REVIEW

**Comment by kratsg** [comment]
@mmilesi - an extra question:

- why is AsgTools added as a dependency?

**Comment by mmilesi** [comment]
In the Makefile.RootCore you mean?

**Comment by kratsg** [comment]
See https://github.com/UCATLAS/xAODAnaHelpers/pull/118/files#diff-6297f591c84d349624bf656eef2d2b01

**Comment by mmilesi** [comment]
Oh right, now I got your point.
Basically I have started implementing and testing in 2.3.13, where the asg::ToolStore::contains() method is available only when explicitly checking out a tag of AsgTools - hence the line in checkoutASGtags.py

In 2.3.14 that tag is in by default. The only issue I see is that this version of the code won't be backward compatible with 2.3.13 unless we keep the AsgTools tag in the list.

**Comment by kratsg** [comment]
Ok, so I figured out why. I had removed everything prior to 2.3.14 since we bumped the official version of the script to 2.3.14+.

**Comment by mmilesi** [comment]
Ciao @gfacini ,

did you get any time for looking at this? 

I understand this is a lot of new stuff pushed into xAH, but being high priority for my analysis (and not only for mine, I guess...) I'd really appreciate you having a "bird's-eye" look at it :)

Very sorry to put so much pressure! I'd go ahead and merge it myself, but this is really not how things should be done...

I extend the "invitation" to other analysts: @johnda102 , @JamesSaxon, @jdandoy .

Thanks very much in advance!

Marco

File: Root/ElectronSelector.cxx [hunk 1]
```diff
@@ -343,31 +342,48 @@ EL::StatusCode ElectronSelector :: initialize ()
...
 
   // if not using LH PID, make sure all the decorations will be set ... by choosing the loosest WP!
+  //
   std::string likelihoodWP = ( m_doLHPIDcut ) ? m_LHOperatingPoint : "VeryLoose";
-  m_el_LH_PIDManager = new ElectronLHPIDManager( likelihoodWP );
+  m_el_LH_PIDManager = new ElectronLHPIDManager( likelihoodWP, m_debug );
+  
+  if  ( m_doLHPIDcut ) {
+    if ( m_debug ) { 
+       Info("initialize()", "Cutting on Electron Likelihood PID! \n ********************" );
```
Comment (gfacini) [inline Root/ElectronSelector.cxx]: Nice! Can even be done without a check of the debug flag.

**Comment by gfacini** [comment]
Looks good! Lots of nice work =)

**Comment by mmilesi** [comment]
@gfacini  :heart:

**Pull request merged**